lib/pengine/native.c

```diff
@@ -1052,9 +1052,10 @@ native_resource_state(const pe_resource_t * rsc, gboolean current)
  *
  * \param[in]  rsc      Resource to check
  * \param[out] list     List to add result to
- * \param[in]  current  0 = where known, 1 = running, 2 = running or pending
+ * \param[in]  current  0 = where allocated, 1 = where running,
+ *                      2 = where running or pending
  *
- * \return If list contains only one node, that node
+ * \return If list contains only one node, that node, or NULL otherwise
```

Comment:
This is why I prefer to use enums rather than magic ints. :) This function signature is public API so we can't change that now in this case ...

Reply:
Agreed. 

Most of the existing calls still use gboolean FALSE/TRUE. I thought about changing those to 0/1 for clarity. But I skipped that for now, because a couple of them come from a gboolean argument to the caller function and I couldn't be bothered.

Reply:
You did change the signature from gboolean to int though. Couldn't you change it from an int to an enum? It would continue to accept 0, 1, and 2, just like it continues to accept FALSE/TRUE now.

Reply:
As a practical matter we probably could, but the difference is that glib guarantees that gboolean is an int, whereas the C standard allows compilers to use an integer type of any sufficient width as enum. It would probably be fine but I'd rather wait until we're breaking API compatibility more widely.